tests/unit/bokeh/core/test_json_encoder.py

```diff
@@ -84,27 +74,24 @@ def test_json_encoder():
   0,
   1,
   128,
-  {{
+  {
     "type": "number",
     "value": "nan"
-  }},
-  {{
-    "type": "map",
-    "entries": [
-      [
-        "key_0",
-        {{
-          "type": "bytes",
-          "data": {{
-            "id": "{rep0.buffers[0].id}"
-          }}
-        }}
-      ]
-    ]
-  }}
+  }
 ]\
 """
 
+def test_json_encoder_bytes():
+    val = {"key": b"uvw"}
+    rep = Serializer().serialize(val)
+    assert rep.buffers is not None and len(rep.buffers) == 1
+
+    encoded_bytes = b64encode(gzip.compress(val["key"], mtime=0)).decode("utf-8")
+
+    assert serialize_json(rep.content) == f"""\
+{{"type":"map","entries":[["key",{{"type":"bytes","data":"{encoded_bytes}"}}]]}}\
```

Comment:
Noting that I split the binary content out into its own test. Trying to round-trip in the "mega test" just became unreadable. The big test can continue to check all the features of the serializer at the top level (content, pretty-printing, etc) and this test can maintain the basic encoding for bytes.